crawler/crawlutils.py

```diff
@@ -458,30 +459,30 @@ def snapshot(
     signal.signal(signal.SIGHUP, signal_handler_exit)
 
     if crawlmode == Modes.OUTCONTAINER:
+	#Use custom filter logic to distribute already running containers
         containers = get_filtered_list_of_containers(options, namespace,
                                                      runtime_env)
 
+    eventQ = options['partition_strategy']['args']['eventQ']
+    delList = options['partition_strategy']['args']['delList']
+
+    if frequency < 0:
+	wait_period = 5
+    else:
+	wait_period = 5
+	
     # This is the main loop of the system, taking a snapshot and sleeping at
     # every iteration.
 
     while True:
-
-        snapshot_time = int(time.time())
-
         if crawlmode == Modes.OUTCONTAINER:
-
-            curr_containers = get_filtered_list_of_containers(
```

Comment:
It might be hard to make it a plugin at the moment. The only pluggable functionality are the methods of this class: https://github.com/cloudviz/agentless-system-crawler/blob/master/crawler/runtime_environment.py